### PR TITLE
fix: remove `NumberPad` callbacks

### PR DESCRIPTION
**Pull request by ethan-tbd**
closes https://github.com/TBD54566975/didpay/issues/22

https://github.com/TBD54566975/didpay/assets/125412902/3f842cfd-6492-491e-a5ff-15ea0f59ba29

https://github.com/TBD54566975/didpay/assets/125412902/8fb6ebd4-3c6c-46dc-a801-929cbe10f557

### CODE REVIEW

File: frontend/lib/shared/number_pad.dart [hunk 1]
```diff
@@ -74,10 +69,18 @@ class NumberPadKey extends HookWidget {
...
         keySize.value = defaultFontSize;
-        (title == '<') ? onDeletePressed() : onKeyPressed(title);
+
+        title == '<'
+            ? _onDeletePressed(enteredAmount)
+            : (title == '.' && enteredAmount.value.contains('.'))
+                ? null
+                : !RegExp(r'\.\d{2}$').hasMatch(enteredAmount.value)
+                    ? _onKeyPressed(title, enteredAmount)
+                    : null;
```
Comment (ethan-tbd) [inline frontend/lib/shared/number_pad.dart:81]: not really happy with how this looks, potentially change to if else blocks?

Comment (wesbillman) [reply frontend/lib/shared/number_pad.dart:81]: I think maybe `if`s with early returns would look better here as well
```dart
if (title == '<') {
  //deal with title
  return;
}
```

File: frontend/lib/shared/number_pad.dart [hunk 2]
```diff
@@ -52,13 +49,11 @@ class NumberPad extends HookWidget {
 
 class NumberPadKey extends HookWidget {
   final String title;
-  final Function(String) onKeyPressed;
-  final VoidCallback onDeletePressed;
+  final ValueNotifier<String> enteredAmount;
```
Comment (wesbillman) [inline frontend/lib/shared/number_pad.dart:52]: I wonder if a `ValueNotifier` is the best approach. Honestly not sure what would be best.

We could also just have like an `initialValue = 0` and a callback with `valueChanged`. Then the users of NumberPad could just hook into the callback. Whatcha think?

Comment (wesbillman) [reply frontend/lib/shared/number_pad.dart:52]: It also feels like the thing that is sent back from NumberPad should be a "number" like `double` vs. a `string`

**Review by wesbillman — APPROVED** [review]

**Review by wesbillman — APPROVED** [review]
LGTM! Nice work! 🥳

**Pull request merged**